src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -814,4 +806,67 @@ private void handleTabletSplit(CDCErrorException cdcErrorException,
             addTabletIfNotPresent(tabletPairList, pair, tableId, offsetContext, schemaNeeded);
         }
     }
+
+    /**
+     * Get the children tablets of the specified tablet which has been split. This API will retry
+     * until the retry limit has been hit.
+     * @param ybTable the {@link YBTable} object to which the split tablet belongs
+     * @param streamId the DB stream ID being used for polling
+     * @param tableId table UUID of the table to which the split tablet belongs
+     * @param splitTabletId tablet UUID of the split tablet
+     * @return {@link GetTabletListToPollForCDCResponse} containing the list of child tablets
+     * @throws Exception when the retry limit has been hit or the metronome pause is interrupted
+     */
+    private GetTabletListToPollForCDCResponse getTabletListResponseWithRetry(
+        YBTable ybTable, String streamId, String tableId, String splitTabletId) throws Exception {
+        short retryCount = 0;
+        while (retryCount <= connectorConfig.maxConnectorRetries()) {
+            try {
+                // Note that YBClient also retries internally if it encounters an error which can
+                // be retried.
+                GetTabletListToPollForCDCResponse response =
+                    this.syncClient.getTabletListToPollForCdc(
+                        ybTable,
+                        streamId,
+                        tableId,
+                        splitTabletId);
+
+                if (response.getTabletCheckpointPairListSize() != 2) {
+                    LOGGER.warn("Found {} tablets for the parent tablet {}",
+                            response.getTabletCheckpointPairListSize(), splitTabletId);
+                    throw new Exception("Found unexpected ("
+                            + response.getTabletCheckpointPairListSize()
+                            + ") number of tablets while trying to fetch the children of parent "
+                            + splitTabletId);
+                }
+
+                retryCount = 0;
+                return response;
+            } catch (Exception e) {
+                ++retryCount;
```

Comment:
Should this code check for an exception type for retry or always retry>

Reply:
We have noticed during nemesis tests that the exceptions thrown are not specific, sometimes it's `NonRecoverableException`, sometimes `NotFound` so that's why we wanted to retry always.